lib/datadog/tracing/contrib/sinatra/patcher.rb

```diff
@@ -42,18 +42,22 @@ def setup_middleware(*args, &block)
         module Patcher
           include Contrib::Patcher
 
+          PATCH_ONLY_ONCE = Core::Utils::OnlyOnce.new
+
           module_function
 
           def target_version
             Integration.version
           end
 
           def patch
-            require_relative 'tracer'
-            register_tracer
+            PATCH_ONLY_ONCE.run do
+              require_relative 'tracer'
+              register_tracer
 
-            patch_default_middlewares
-            setup_tracer
+              patch_default_middlewares
+              setup_tracer
+            end
           end
```

Comment:
Hmm... I actually think this is a bit of a no-op, compared to what we're doing now?

E.g. since we're still monkey patching the sinatra classes with the `TracerSetupPatch`, we'll still be calling `Datadog.configure` multiple times for each sinatra app.

I don't quite see how this moves the needle from where we are right now :thinking: